src/types/block/mod.rs

```diff
@@ -3,50 +3,102 @@ pub mod redstone;
 
 use self::{entity::BlockEntity, redstone::Component};
 
+/// Base trait for a block
```

Comment:
Specify all blocks and what to do w/ block entities and components and how to construct this